### PR TITLE
[SDPT-780]-Enhance NiFi Connector to support SDP

### PR DESCRIPTION
**Pull request by chennv4**
**Problem:**
Existing version supports only pravega

**Solution:**
Enhanced NiFi Connector to support SDP

**Test Results**
Deployed on DellIt digital team cluster and validated.

### CODE REVIEW

File: nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java [hunk 1]
```diff
@@ -127,6 +128,23 @@ public ValidationResult validate(String subject, String input, ValidationContext
...
+            .name("keycloakJson")
+            .displayName("Keycloak Json")
+            .description("The Keycloak Json.")
+            .required(false)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor PROP_LOCAL_PRAVEGA = new PropertyDescriptor.Builder()
+            .name("PROP_LOCAL_PRAVEGA")
+            .displayName("Local Pravega (true/false)")
```
Comment (claudiofahey) [inline nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java]: Rename this to "Create Scope" and PROP_CREATE_SCOPE. here and throughput this PR. Default should be true.

File: nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java [hunk 2]
```diff
@@ -144,10 +164,9 @@ public ValidationResult validate(String subject, String input, ValidationContext
     public ClientConfig getClientConfig(final ProcessContext context) {
         try {
             final URI controllerURI = new URI(context.getProperty(PROP_CONTROLLER).getValue());
-            final ClientConfig clientConfig = ClientConfig.builder()
-                    .controllerURI(controllerURI)
-                    .build();
-            return clientConfig;
+            final ClientConfig.ClientConfigBuilder clientBuilder = ClientConfig.builder().controllerURI(controllerURI);
+            clientBuilder.credentials(new PravegaKeycloakCredentialsFromString(context.getProperty(PROP_KEYCLOAK_JSON).getValue()));
```
Comment (claudiofahey) [inline nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java]: Need to handle case when Keycloak JSON is not provided.

File: nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/ConsumePravega.java [hunk 3]
```diff
@@ -243,6 +245,7 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
             logger.info("onTrigger: ProcessorNotReadyException", new Object[]{e});
             context.yield();
         } catch (final Exception e) {
+            e.printStackTrace();
```
Comment (claudiofahey) [inline nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/ConsumePravega.java]: Replace with `logger.error("Error", e)`. Same with below.

**Review by claudiofahey — CHANGES_REQUESTED** [review]

File: nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java (hunk 1 above)
Comment (chennv4) [reply nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java]: Changed

File: nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java (hunk 2 above)
Comment (chennv4) [reply nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java]: Good cache. Thanks
Implemented

File: nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/ConsumePravega.java (hunk 3 above)
Comment (chennv4) [reply nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/ConsumePravega.java]: done

File: nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java [hunk 4]
```diff
@@ -136,18 +154,36 @@ public ValidationResult validate(String subject, String input, ValidationContext
...
-                    .controllerURI(controllerURI)
-                    .build();
+            final ClientConfig.ClientConfigBuilder clientBuilder = ClientConfig.builder().controllerURI(controllerURI);
+            if(createScope)
+            {
+                clientConfig = ClientConfig.builder()
+                        .controllerURI(controllerURI)
+                        .build();
+            }
+            else if(!createScope && (context.getProperty(PROP_KEYCLOAK_JSON).getValue() != null))
```
Comment (claudiofahey) [inline nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java]: What does `createScope` have to do with authentication? Just use `context.getProperty(PROP_KEYCLOAK_JSON).getValue() != null` to determine whether to use Keycloak. Do not use the `createScope` flag here.

**Review by claudiofahey — CHANGES_REQUESTED** [review]

File: nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java (hunk 4 above)
Comment (chennv4) [reply nifi-pravega-processors/src/main/java/org/apache/nifi/processors/pravega/AbstractPravegaProcessor.java]: removed the createscope check

**Review by claudiofahey — APPROVED** [review]

**Pull request merged**